### ISSUE TITLE
Missing Definition of Timestamp Predicate

### ISSUE THREAD

**Issue by greenTara** [streamreasoning/RSP-QL#26]
In a timestamped graph, the predicate in the timestamp triple cannot be just any predicate.
It should have a specified range (of termporal entities).
This can be accomplished, to some extent, with RDFS, by identifying an rdfs:range.
Also the (partial or total) order that will be used to structure a stream must be specified.

**Comment by greenTara** [comment]
There are already some predicates defined in external standards, e.g. PROV-O (https://www.w3.org/TR/prov-o/), that intuitively appear to be suitable as timestamp predicates. Let's consider these as examples
- https://www.w3.org/TR/2013/REC-prov-o-20130430/#generatedAtTime
  Here's the PROV-O definition

```
:generatedAtTime
    a owl:DatatypeProperty ;
    rdfs:comment "The time at which an entity was completely created and is available for use."@en ;
    rdfs:domain :Entity ;
    rdfs:isDefinedBy <http://www.w3.org/ns/prov-o#> ;
    rdfs:label "generatedAtTime" ;
    rdfs:range xsd:dateTime ;
    :category "expanded" ;
    :component "entities-activities" ;
    :editorialNote "It is the intent that the property chain holds: (prov:qualifiedGeneration o prov:atTime) rdfs:subPropertyOf prov:generatedAtTime."@en ;
    :qualifiedForm :Generation, :atTime 
```

Notice that this is a subproperty of owl:DatatypeProperty. In OWL-DL properties must be either a datatype property or an object property - the range of a property cannot contain both rdf:Literal values and owl:Thing objects, and making such a statement causes the ontology to be inconsistent. Deviating from this convention should not be done lightly, as it has a big impact on applicability of reasoning engines.

The missing piece to make prov:generatedAtTime a timestamp predicate is to define the partial order.
Option1. Define an rdf:Property for the partial order.
Suppose that we define a property
rsp:beforeOrSameXSDDateTime
that has rdfs:domain and rdfs:range to be the value space of xsd:dateTime. 

```
rsp:beforeOrSameXSDDateTime 
  a rsp:TemporalPartialOrder;
  rdfs:domain xsd:dateTime;
  rdfs:range xsd:dateTime
```

(Note: this property could not be reasoned with in OWL-DL because of the domain containing rdf:Literals)
and then specify what the extension of this property is intended to be, in terms of the value space of xsd:dateTime (being careful about values with and without timezones).
This property could be used by multiple timestamp predicates.
The connection between a timestamp predicate and a partial order could be formalized with a property

```
prov:generatedAtTime 
  a rsp:TimestampProperty;
  rsp:usesTemporalPartialOrder rsp:beforeOrSameXSDDateTime
```

A conforming RSP engine would not accept or generate streams that are not valid w.r.t. these specifications.
Addtional supporting artifacts:
- formalization of rsp:TimestampProperty, rsp:TemporalPartialOrder, rsp:usesTemporalPartialOrder, ... in an ontology (probably OWL Full or higher, e.g. ISO Common Logic)

**Comment by greenTara** [comment]
This issue is being addressed in the pull request #67 because much of what was needed to define timestamp predicates was already in the document, it just needed to be pulled together into one place.

**Comment by greenTara** [comment]
resoved by merge of #67
